web/js/layers/wv.layers.modal.js

```diff
@@ -120,458 +140,350 @@ wv.layers.modal = wv.layers.modal || function(models, ui, config) {
-      $categories.show()
-        .isotope();
-      $nav.show();
+    // If URL parameter is set, draw that type of modal view.
+    if (config.parameters.modalView == 'categories') {
+      crumbText = 'Categories';
+      drawCategories();
+    } else if (config.parameters.modalView == 'measurements') {
+      crumbText = 'Measurements';
+      drawAllMeasurements();
+    } else if (config.parameters.modalView == 'layers') {
+      crumbText = 'Layers';
+      drawAllLayers();
 
+      // Else set the default views per projection.
+    } else if (projection == 'geographic') {
+      crumbText = 'Categories';
+      drawCategories();
     } else {
-      filter();
+      crumbText = 'All Measurements';
+      drawAllMeasurements();
     }
-
-    redoScrollbar();
   };
 
-  var resize = function() {
-    if ($(self.selector)
-      .dialog("isOpen")) {
-      redo();
+  var drawCategories = function() {
+    $categories.empty();
+    if ($categories.data('isotope')) {
+      $categories.isotope('destroy');
     }
-  };
-
-  // If this is not the geographic projection, All layers are always drawn
-  // and filtered, so thats the default page. Dont show breadcrumb
-  var checkSearch = function() {
-    if (searchBool) {
-      var crumbText;
-      if (models.proj.selected.id !== 'geographic') {
-        crumbText = 'Layers';
-      } else {
-        crumbText = 'Categories';
-      }
-      var $homeCrumb = $('<a></a>')
-        .text(crumbText)
-        .attr('alt', crumbText)
-        .attr('title', 'Back to ' + crumbText)
-        .click(showDefaultPage);
+    $allLayers.hide();
+    $nav.empty();
 
-      $breadcrumb.append($homeCrumb)
-        .append('<span> / Search Results</span>');
+    _.each(config.categories, function(metaCategory, metaCategoryName) {
 
-      $allLayers.prepend($breadcrumb);
-      $('#layers-search-input')
-        .show();
+      _.each(config.categories[metaCategoryName], function(category, name) {
+        var sortNumber;
 
-      $('label.search-icon')
-        .addClass('search-on');
-      $('label.search-on')
-        .click(showDefaultPage);
+        // Check if categories have settings with the same projection.
+        hasSetting = false;
+        _.each( category.measurements, function( measurement, index ) {
+          var projection = models.proj.selected.id;
+          var current = config.measurements[measurement];
+          checkSourceSettings(current);
+        });
 
-      $breadcrumb.show();
-    }
-  };
+        if (hasSetting === true) {
+          if (category.placement) {
+            if (category.placement === 'first') {
+              sortNumber = 1;
+            } else if (category.placement === 'last') {
+              sortNumber = 3;
+            }
+          } else {
+            sortNumber = 2;
+          }
+          var $category = $('<div></div>').attr('data-sort', sortNumber).addClass('layer-category layer-category-' + interestCssName(metaCategoryName)).attr('id', category.id);
```

Comment:
You can create a new element with the attributes you need, rather than chaining extra function calls on the end;
```
$('<div />', {
  class: 'layer-category layer-category-' + interestCssName(metaCategoryName)),
  id: category.id,
  'data-sort': sortNumber
});
```
It's more readable, and more efficient that way.